tests/lib/rules/no-property-in-node.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const { RuleTester } = require('@typescript-eslint/rule-tester');
+const path = require('path');
+const rule = require('../../../lib/rules/no-property-in-node');
+
+RuleTester.afterAll = after;
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: './tsconfig.json',
+    tsconfigRootDir: path.join(__dirname, 'fixtures'),
+  },
+});
+
+ruleTester.run('no-property-in-node', rule, {
+  invalid: [
+    {
+      code: `
+        import { TSESTree } from '@typescript-eslint/utils';
+        declare const node: TSESTree.Node;
+        'a' in node;
+      `,
+      errors: [
+        {
+          column: 9,
+          line: 4,
+          endColumn: 20,
+          endLine: 4,
+          messageId: 'in',
+        },
+      ],
+    },
+    {
+      code: `
+        import { TSESTree } from '@typescript-eslint/utils';
+        type Other = { key: true };
+        declare const node: TSESTree.Node | Other;
+        'a' in node;
+      `,
+      errors: [
+        {
+          column: 9,
+          line: 5,
+          endColumn: 20,
+          endLine: 5,
+          messageId: 'in',
+        },
+      ],
+    },
+    {
+      code: `
+        import * as ESTree from "estree";
+        declare const node: ESTree.Node;
+        'a' in node;
+      `,
+      errors: [
+        {
+          column: 9,
+          line: 4,
+          endColumn: 20,
+          endLine: 4,
+          messageId: 'in',
+        },
+      ],
+    },
+    {
+      code: `
+        import * as eslint from 'eslint';
+        const listener: eslint.Rule.RuleListener = {
+          ArrayExpression(node) {
+            'a' in node;
+          },
+        };
+      `,
+      errors: [
+        {
+          column: 13,
+          line: 5,
+          endColumn: 24,
+          endLine: 5,
+          messageId: 'in',
+        },
+      ],
+    },
+  ],
+  valid: [
```

Comment:
Heh, yeah, I think I'm the only person who does this. It's because I use `eslint-plugin-perfectionist` to sort everything alphabetically in my repos. I'll switch 🙂

Reply:
Well, you're not the only one 😄 
https://github.com/microsoft/rushstack/blob/fe1cbf4e3f325e4d55c4c02316715999b72b4d20/eslint/eslint-plugin/src/test/typedef-var.test.ts

(an example I saw yesterday and had the same thought as bmish)